ocaml/parsing/ast_iterator.ml

```diff
@@ -969,8 +972,12 @@ let default_iterator =
       this.location this a.attr_loc
     );
     attributes = (fun this l -> List.iter (this.attribute this) l);
-    (* [ast_iterator] should not know about the structure of mode expressions *)
-    modes = (fun _this _m -> ());
+    (* Location inside a mode expression needs to be traversed. *)
+    modes = (fun this m ->
+      let iter_const sub : ME.Const.t -> _ =
+        fun m -> iter_loc sub (m : ME.Const.t :> _ Location.loc)
+      in
+      iter_loc_txt this (fun sub -> List.iter (iter_const sub)) m);
```

Comment:
Is it possible to move this logic to the client code of `ast_iterator`? I think in the future `Mode_expr.t` will be more complex than a list (it will be a real "expression" with mode operators join, meet, etc.) and it would occur strange to me if the `ast_iterator` needs to know about that.